sway/tree/view.c

```diff
@@ -522,6 +555,17 @@ void view_map(struct sway_view *view, struct wlr_surface *wlr_surface) {
 
 	container_damage_whole(cont);
 	view_handle_container_reparent(&view->container_reparent, NULL);
+
+	if (prev_focus != focus) {
+		struct sway_container *ws = workspace_by_name(prev_ws);
+		if (!ws) {
+			ws = workspace_create(prev_output, prev_ws);
+		}
+		input_manager_set_focus(input_manager, ws);
+		workspace_switch(ws);
```

Comment:
That will really do the switch and a transaction, then when `view_map()` ends it will do a second `arrange_and_commit`... then as you pointed out on irc if this was assigned through a rule the focus is changed back again in a 3rd transaction?

Actually do we ever want to switch to the workspace where a new view got created? That happens when matching parent's pid?
if e.g. firefox ever wants to open a popup in my back when I'm typing something in the terminal I sure wouldn't want switching to firefox's workspace...

Reply:
@SirCmpwn You might want to consider waiting until atomic is merged then rebasing this.

Reply:
`workspace_switch(ws)` still looks here, doesn't add up with your (outdated) comment?

Reply:
Derp